include/rocksdb/slice.h

```diff
@@ -116,6 +118,52 @@ class Slice {
   // Intentionally copyable
 };
 
+class PinnableSlice : public Slice, public Cleanable {
+ public:
+  PinnableSlice() {}
+
+  inline void PinSlice(const Slice& s, CleanupFunction f, void* arg1,
+                       void* arg2) {
+    data_ = s.data();
+    size_ = s.size();
+    RegisterCleanup(f, arg1, arg2);
+  }
+
+  inline void PinSlice(const Slice& s, Cleanable* cleanable) {
+    data_ = s.data();
+    size_ = s.size();
+    cleanable->DelegateCleanupsTo(this);
+  }
+
+  inline void PinHeap(std::string* s) {
```

Comment:
I think we should not have code that is not used, it makes people wonder (where is this code used) ?
I would prefer if we remove it, but this is my personal opinion, It's your call